### PR TITLE
Add Path annotation based operation name provider (#114)

### LINKED ISSUE #114: Incorrect operation name in WildcardOperationName when path value is the same
https://github.com/opentracing-contrib/java-jaxrs/blob/master/opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java#L74
```
@GET
@Path("/test/{id}/{subid}")
```
When the path parameters in the given URI have same value such as: `/test/1/1`, the operation name will be matched as `/test/{subid}/{subid}`

### PR DESCRIPTION
**Pull request by auke-**
An alternative for the WildcardOperationName implementation (fixes #114).

### CODE REVIEW

**Comment by pavolloffay** [comment]
thanks for the pr @auke-, could you please fix ci to pass?

**Comment by auke-** [comment]
> thanks for the pr @auke-, could you please fix ci to pass?

I tried to understand why the ci is failing, but from what i can see it seems unrelated to my change.

File: opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java [hunk 1]
```diff
@@ -103,4 +104,39 @@ public static Builder newBuilder() {
...
     }
   }
+
+  /**
+   * As operation name provides the path annotation from both the class and the method e.g:
+   *
+   * resource class annotated with @Path("/foo/") and method annotated with
+   * @Path("bar/{name: \\w+}") produces "/foo/bar/{name}"
+   */
+  class PathAnnotationOperationName implements OperationNameProvider {
```
Comment (pavolloffay) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java]: Could you please move this implementation to `WildcardOperationName` and provide a test?

Comment (auke-) [reply opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java]: I've might been a bit to quick and copied the javadoc. My solution doesn't strip the patterns from the variables, it just uses the plain value of the path annotation.

Do you know a solution for resolving variable names independent from the JAX-RS implementation?

**Comment by pavolloffay** [comment]
@auke- I have submitted #119 to address my concerns. The idea is to keep only one operation name provider for wildcard path.

**Comment by auke-** [comment]
Sounds good to me, thanks!

**Pull request closed without merging**